docs/config-and-instances.md

```diff
@@ -0,0 +1,340 @@
+'snapshot' of a `VM` object, whenever it is getting created.
+The flag `instanceConfig` can be set on a `Config`, to tell the system
+to create a new `Config` object whenever a VM is started based on this
+`Config`.
+
+By default this `Config` is only created when the VM is launched. Thus changes
+performed on the `VM` object at runtime will not be reflected in the created
+`Config`. But the switch `autoSave` can be used to automatically propagate
+changes done to the `VM` object at runtime, to the `Config` object.
+
+But let's look at an example to grasp this:
+
+**Note:** The only relevant endpoints for the initial concept are `/start` and
+`/stop`. Additional endpoints (`/save` and `/snapshot`) are just to illustrate
+how the mechanism could eventually be used for other functionality as well.
+
+```
+$c1        $c2        $c3        $i1        $i2        $i3       $i4
+POST
+/start                           POST
+           POST
+/stop                            DELETE
+           /start                           POST
+           /snapshot  POST
+           /save PATCH
+           /stop                            DELETE
+           /start                                      POST
+           PATCH                    ←                  PATCH
+           PATCH
+PATCH
+/start
+                                                                 POST
+/stop                                                            DELETE
+```
+
+**Note:** The `/snapshot` call is just an example how this mechanism could
+be used for manually triggering Config snapshots. It's not a required element.
+
+**Note:** The `/save` call is also just an example, to show that the mechanism
+can also be used to save the current runtime state (in the case above $i2) by
+merging it back into the config ($c2).
+
+1. `POST` $c1 config
+2. `GET` $c1/start
+   → creates instance `POST` $i1
+   → create `POST` $c2 (with instanceConfig and autoSave) from $i1 with
+     expanded domain
+3. `GET` $c1/stop
+   → `DELETE` $i1
+4. `GET` $c2/start
+   → creates instance `POST` $i2
+   → updates (not .state) `PATCH` $c2 from $i2
+5. `GET` $c2/snapshot
+   → creates `POST` $c3 (without autoSave) from $i2
+6. `GET` $c2/save
+   → updates (not state) `PATCH` $c2 from $i2
+7. `GET` $c2/stop
+   → DELETE $i2
+8. `GET` $c2/start
+   → creates instance POST $i3
+   → updates (not .state) `PATCH` $c2 from $i3
+9. `PATCH` $i3
+   → hot-adds device to domain
+   → updates (not .state) `PATCH` $c2 from $i3
+10. `PATCH` $c3
+11. `PATCH` $c1 (`instanceConfig: False`)
+12. `GET` $c1/start
+   → creates instance `POST` $i4
+12. `GET` $c1/stop
+   → deletes instance `DELETE` $i4
+
+
+### Observations
+* Expansion is performed at runtime
+* Imperative on config, declarative on instance
+* Hot-plugging by modifying the instance
+* The dynamic instance data can be reflected in the VM.state.domain fields
+* A single `Config` can be used to create several instances. Or phrased
```

Comment:
meh - Alright that there are different opinions, let's see what further thoughts on the topic will bring.

By itself it would not have been pools, it is rather intended to be a mechanism to _support_ pools.
Like (the discussed _idea_) of presets was a mechanism to support instance types.